crates/subspace-archiving/src/archiver.rs

```diff
@@ -647,17 +648,20 @@ impl Archiver {
                                 + 1
                                 + Compact::compact_len(&(bytes.len() as u32))
                                 + block_object.offset() as usize;
-                            let offset = (offset_in_segment % RawRecord::SIZE).try_into().expect(
-                                "Offset within piece should always fit in 16-bit integer; qed",
-                            );
+                            let raw_offset =
+                                (offset_in_segment % RawRecord::SIZE).try_into().expect(
+                                    "Offset within piece should always fit in 32-bit integer; qed",
+                                );
 
                             if let Some(piece_object_mapping) = corrected_object_mapping
                                 .get_mut(offset_in_segment / RawRecord::SIZE)
                             {
-                                piece_object_mapping.objects.push(PieceObject::V0 {
-                                    hash: block_object.hash(),
-                                    offset,
-                                });
+                                piece_object_mapping
+                                    .objects
+                                    .push(PieceObject::from_raw_offset(
+                                        block_object.hash(),
+                                        raw_offset,
+                                    ));
```

Comment:
I do not understand why it is helpful to even expose that there is something called raw offet :thinking: 

Pieces do not store raw records, so raw offset is irrelevant here and was used by mistake. When you do transition to raw records during data retrieval, you'll need to do data transformations and that is where you'll also convert offsets accordingly. But as far as global object are concerned in general there is just one kind of offset IMO.

Reply:
In this code, I can convert to regular offsets before creating the struct.

In the tests, I didn't know how to use regular offsets. I tried a few different ways, but it didn't work. That might have been because of a bug in the offset calculations, so I'll try again.

Reply:
You didn't specify what you mean by "didn't work", but I suspect you meant trying to get the data from a piece by offset results in the wrong data. That is true, you'll have to extract record and convert it to raw record or account for zero bytes during retrieval in some other way to get the correct result.

Reply:
The tests already account for the zero bytes, so I just need to convert from regular to raw offsets.